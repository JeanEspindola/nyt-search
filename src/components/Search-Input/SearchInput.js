/*
* Search Input Presentation Component
*
* Displays the search input bar. The search action is in onSubmit action.
*
* */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form, FormGroup, FormControl, InputGroup, Glyphicon, ControlLabel } from 'react-bootstrap';
import './SearchInput.css';

const page = 0;

class SearchInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      query: '',
    };

    this.onChangeQuery = this.onChangeQuery.bind(this);
    this.onSearch = this.onSearch.bind(this);
  }

  onChangeQuery(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSearch(e) {
    e.preventDefault();

    const { query } = this.state;

    if (query === '') {
      return;
    }
    this.props.onQuerySearch(query, page);
  }

  render() {
    const { query } = this.state;

    return (
      <Form className="Search-Form" onSubmit={this.onSearch}>
        <FormGroup>
          <ControlLabel>Type search query term in here:</ControlLabel>
          <InputGroup>
            <FormControl
              type="text"
              placeholder="Search your terms here."
              value={query}
              onChange={this.onChangeQuery}
              name="query"
            />
            <InputGroup.Addon>
              <Glyphicon glyph="search" />
            </InputGroup.Addon>
          </InputGroup>
        </FormGroup>
      </Form>
    );
  }
}

SearchInput.propTypes = {
  onQuerySearch: PropTypes.func.isRequired,
};

export default SearchInput;
