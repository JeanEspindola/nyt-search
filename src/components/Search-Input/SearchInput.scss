
.Search-Form {
  margin-top: 20px;
}
