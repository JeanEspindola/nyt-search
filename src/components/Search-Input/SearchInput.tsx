/*
* Search Input Presentation Component
*
* Displays the search input bar. The search action is in onSubmit action.
*
* */
import React, { useState } from 'react'
import { Form, FormControl, FormGroup, FormLabel, InputGroup } from 'react-bootstrap'
import * as Icon from 'react-bootstrap-icons'
import { useDispatch } from 'react-redux'
import { loadSearchResults } from '../../redux/actions/articleSearchAction'
import styles from './SearchInput.module.scss';
import { FormattedMessage, useIntl } from 'react-intl'

const SearchInput = () => {
  const dispatch = useDispatch()
  const [query, setQuery] = useState('')

  const intl = useIntl();

  const placeholderText = intl.formatMessage({ id: 'searchTerms' });

  const onChangeQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
    setQuery(e.target.value)
  }

  const onSearch = (e: React.BaseSyntheticEvent) => {
    e.preventDefault();

    if (query === '') {
      return;
    }

    dispatch(loadSearchResults(query, 0))
  }

  return (
    <Form className={styles.container} onSubmit={onSearch}>
      <FormGroup controlId="formSearch">
        <FormLabel>
          <FormattedMessage id={'typeQuery'} />
        </FormLabel>
        <InputGroup className="mb-3">
          <FormControl type="text" placeholder={placeholderText}
            value={query}
            onChange={onChangeQuery}
            name="query"
          />
          <InputGroup.Append>
            <InputGroup.Text id="basic-addon2"><Icon.Search /></InputGroup.Text>
          </InputGroup.Append>
        </InputGroup>
      </FormGroup>
    </Form>
  )
}

export default SearchInput
