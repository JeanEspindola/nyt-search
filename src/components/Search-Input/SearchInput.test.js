import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchInput from './SearchInput';

describe('SearchInput', () => {
  const mockOnQuerySearch = jest.fn;

  let Component;

  beforeEach(() => {
    Component = shallow(<SearchInput onQuerySearch={mockOnQuerySearch} />);
  });

  it('renders SearchInput without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });
});
