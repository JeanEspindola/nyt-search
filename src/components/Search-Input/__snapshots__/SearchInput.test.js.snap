// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders SearchInput without crashing 1`] = `
<Form
  bsClass="form"
  className="Search-Form"
  componentClass="form"
  horizontal={false}
  inline={false}
  onSubmit={[Function]}
>
  <FormGroup
    bsClass="form-group"
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      Type search query term in here:
    </ControlLabel>
    <InputGroup
      bsClass="input-group"
    >
      <FormControl
        bsClass="form-control"
        componentClass="input"
        name="query"
        onChange={[Function]}
        placeholder="Search your terms here."
        type="text"
        value=""
      />
      <InputGroupAddon
        bsClass="input-group-addon"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="search"
        />
      </InputGroupAddon>
    </InputGroup>
  </FormGroup>
</Form>
`;
